akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -268,6 +268,40 @@ private[akka] final case class Grouped[T](n: Int) extends PushPullStage[T, immut
 /**
  * INTERNAL API
  */
+private[akka] final case class Sliding[T](n: Int, step: Int) extends PushPullStage[T, immutable.Seq[T]] {
+  private var buf = immutable.Stack.empty[T]
```

Comment:
I was picking `Stack` because it has O(1) append complexity. Vector has almost the same but depends on certain assumptions (http://docs.scala-lang.org/overviews/collections/performance-characteristics.html). `Vector` should be also ok, sure!

Also I didn't pick a builder since I don't know if you can append to a vector builder for example. If not wouldn't I have to always reverse the vector when I output it?